AppStoreExample/AppStoreExample/Views/AppSyncViewController.swift

```diff
@@ -33,21 +33,29 @@ private extension AppSyncViewController {
         retryButton?.layer.borderColor = UIColor.darkGray.cgColor
         retryButton?.layer.borderWidth = 1.0
         retryButton?.layer.cornerRadius = 3.0
-        retryContainerView?.isHidden = true
     }
     
     func configureBinding() {
-        viewModel.onDidFinishSync = { [weak self] in
-            self?.onDidFinishSync()
+        viewModel.appSyncStateListener = {
```

Comment:
I prefer calling a separate method.. that way you avoid adding guards for self, and simplify the implementation